server/src/services/download.service.ts

```diff
@@ -96,7 +96,8 @@ export class DownloadService {
 
       const { originalPath, originalFileName } = asset;
 
-      let filename = originalFileName;
+      const extension = parse(originalPath).ext;
+      let filename = `${parse(originalFileName).name}${extension}`;
```

Comment:
Why does it have the wrong extension to begin with? I think I would rather fix the root cause instead of patching it everyone it is used.

Reply:
@jrasm91 I agree (sort of). Hence my comments https://github.com/immich-app/immich/pull/8659#issuecomment-2046140774 and https://github.com/immich-app/immich/pull/8659#issuecomment-2072999502.

What is your opinion on that?

Reply:
iOS doesn't have it embedded but Android does. Either way getting the extracted video might be nice when you download it. The name for the motion asset should come from the original when it is uploaded, but that should not include the extension...

Reply:
A filename includes the extension (by definition), which makes it confusing as that does not match what users perceive as such.

Having said that I'd say its fair to transform the (original)filename of files when converted (thus, if I am following, never having the incorrect extension for these records at time of download).

Reply:
Pretty sure the old extension is coming from here:

https://github.com/immich-app/immich/blob/main/server/src/services/metadata.service.ts#L441

If you are going to fix anything, it should be here instead. It is probably possible to write a migration to fix affected files as well. 

IMO we should store the correct filename and extension when the asset is created and then never touch or need to touch/modify if moving forward. The current solution also introduces the possibility of the extension case changing between upload and download.